.github/workflows/release-prepare.yml

```diff
@@ -151,23 +151,21 @@ jobs:
           fi
           git merge --ff-only "${HOLOCHAIN_SOURCE_BRANCH}"
 
-      - name: Restore holochain cargo related state and build files
+      - name: Restore Cargo cache
         if: ${{ inputs.skip_prepare_logic != 'true' }}
-        uses: steveeJ-forks/actions-cache/restore@retry
-        id: restore-build-files
+        uses:  actions/cache@v4
         with:
+          # See https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci
+          # See https://github.com/actions/cache/blob/main/examples.md#rust---cargo
           path: |
-            /var/tmp/holochain_repo/.cargo/bin/
```

Comment:
I assume this ran inside a Nix shell at some point where we force `.cargo` into `$(pwd)/.cargo`. On a `ubuntu-latest` actions runner, it's found in `~/.cargo`. So this has been broken for quite a while and the only thing we've been caching is the `target` directory.

Reply:
see the definition of `HOLOCHAIN_REPO`: https://github.com/holochain/holochain/pull/3764/files#diff-c653f3e33b01c829488a4008bada88d6987346456be9abb31ce6064ea518410aR46

inside the `nix develop` shell, [`CARGO_HOME`](https://github.com/holochain/holochain/blob/2f493b76b3d37bebd27f93f6a5375f6390446dae/nix/modules/devShells.nix#L170) is set relative to that. maybe we've been less pedantic about using `cargo` only within `nix develop` shells so we now also get things installed under `~`.

Reply:
@ThetaSinner pinging as i'm not sure if notifications come in for closed PRs otherwise